Sources/Testing/Issues/Issue.swift

```diff
@@ -114,13 +114,34 @@ public struct Issue: Sendable {
   init(
     kind: Kind,
     comments: [Comment],
-    sourceContext: SourceContext = .init()
+    sourceContext: SourceContext
   ) {
     self.kind = kind
     self.comments = comments
     self.sourceContext = sourceContext
   }
 
+  /// Initialize an issue instance representing a caught error.
+  ///
+  /// - Parameters:
+  ///   - error: The error which was caught which this issue is describing.
+  ///   - sourceLocation: The source location of the issue. This value is used
+  ///     to construct an instance of ``SourceContext``.
+  ///
+  /// The ``sourceContext`` property will have a ``SourceContext/backtrace``
+  /// property whose value is the backtrace for the first throw of `error`.
+  init(
+    forCaughtError error: any Error,
+    sourceLocation: SourceLocation? = nil
```

Comment:
This initializer makes it possible to accidentally produce an `Issue` with no source context at all, because `sourceLocation` defaults to `nil` and the backtrace is not guaranteed to be available.

Reply:
That's already true prior to this PR—for example, in the way `.errorCaught` issues are created in `Runner.Plan`.

Reply:
Right, but in the cases where it happens now, we had to explicitly write it out, right?

Reply:
The common case for caught errors is to attempt to get the error's backtrace, if one is available, else `nil`. And typically caught errors don't include an explicitly-supplied `SourceLocation`. So I think this new convenience initializer strikes the right balance, and the PR overall helps make many things more explicit